include/drjit-core/jit.h

```diff
@@ -1130,6 +1130,16 @@ JIT_INLINE void jit_var_dec_ref(uint32_t index) JIT_NOEXCEPT {
 #define jit_var_inc_ref jit_var_inc_ref_impl
 #endif
 
+/**
+ * \brief Lock the recursive state mutex.
```

Comment:
Add a comment: This can improve performance when a single thread performs a large number of Dr.Jit operations in sequence.